bigframes/core/compile/compiled.py

```diff
@@ -331,18 +335,18 @@ def aggregate(
             if dropna:
                 for column_id in by_column_ids:
                     expr = expr._filter(expr._get_ibis_column(column_id).notnull())
-            # Can maybe remove this as Ordering id is redundant as by_column is unique after aggregation
-            return expr._project_offsets()
+            return expr
         else:
             aggregates = {**stats, ORDER_ID_COLUMN: ibis_types.literal(0)}
             result = table.aggregate(**aggregates)
-            # Ordering is irrelevant for single-row output, but set ordering id regardless as other ops(join etc.) expect it.
+            # Ordering is irrelevant for single-row output, but set ordering id regardless
+            # as other ops(join etc.) expect it.
             # TODO: Maybe can make completely empty
             ordering = ExpressionOrdering(
                 ordering_value_columns=tuple([]),
                 total_ordering_columns=frozenset([]),
             )
-            return OrderedIR(
+            return UnorderedIR(
```

Comment:
This should produce ordered IR , compiler can drop immediately if ordering not needed.

Reply:
Fixed. Done